app/src/main/java/io/gnosis/safe/ErrorHandler.kt

```diff
@@ -77,39 +105,73 @@ enum class Error(
 fun Throwable.toError(): Error =
     when {
         // Common client errors
-        this is UsedSafeAddress -> Error.ERROR_1101
-        this is InvalidAddressException -> Error.ERROR_1102
-        this is InvalidSeedPhrase -> Error.ERROR_1103
-        this is InvalidPrivateKey -> Error.ERROR_1103
-        this is MismatchingSafeTxHash -> Error.ERROR_1104
-        this is SafeNotSupported -> Error.ERROR_1105
-        this is EnsResolutionError -> Error.ERROR_1106
-        this is EnsReverseRecordNotSetError -> Error.ERROR_1107
-        this is EnsInvalidError -> Error.ERROR_1108
-        this is SafeNotFound -> Error.ERROR_1109
-        this is InvalidName -> Error.ERROR_1110
+        this is UsedSafeAddress -> Error.Error1101
+        this is InvalidAddressException -> Error.Error1102
+        this is InvalidSeedPhrase -> Error.Error1103
+        this is InvalidPrivateKey -> Error.Error1103
+        this is MismatchingSafeTxHash -> Error.Error1104
+        this is SafeNotSupported -> Error.Error1105
+        this is EnsResolutionError -> Error.Error1106
+        this is EnsReverseRecordNotSetError -> Error.Error1107
+        this is EnsInvalidError -> Error.Error1108
+        this is SafeNotFound -> Error.Error1109
+        this is InvalidName -> Error.Error1110
 
         // Network-related errors
         this is HttpException -> {
             this.let {
-                when (this.code()) {
-                    HttpCodes.BAD_REQUEST -> Error.ERROR_400
-                    HttpCodes.UNAUTHORIZED -> Error.ERROR_401
-                    HttpCodes.FORBIDDEN -> Error.ERROR_403
-                    HttpCodes.SERVER_ERROR -> Error.ERROR_500
-                    else -> Error.ERROR_UNKNOWN
+                when {
+                    this.code() == HttpCodes.BAD_REQUEST -> Error.Error400
+                    this.code() == HttpCodes.UNAUTHORIZED -> Error.Error401
+                    this.code() == HttpCodes.FORBIDDEN -> Error.Error403
+                    this.code() == HttpCodes.NOT_FOUND -> Error.Error404
+                    this.code() == HttpCodes.SERVER_ERROR -> Error.Error500(500)
+                    this.code() == 422 -> {
+                        val errorBodyString = it.response()?.errorBody()?.string()
+                        val serverError = errorBodyString?.let {
+                            if (errorBodyString.isNotEmpty()) {
+                                serverErrorAdapter.fromJson(errorBodyString)
+                            } else {
+                                null
+                            }
+                        }
+
+                        if (Tracker.isInitialized()) {
```

Comment:
as discussed let's have tracking logic at places where mapping is called. BaseFragment and BaseActivity already have tracker property.

Reply:
Done in: 2a2648867868a5b1437085a01e4898229f1b77cf